Lib/test/test_pdb.py

```diff
@@ -1417,7 +1417,7 @@ def test_blocks_at_first_code_line(self):
     def test_relative_imports(self):
```

Comment:
I think we do want to explicitly test both paths through `runpy._get_module_details` here, as otherwise we're exposed to an internal refactoring in `runpy` breaking the above assumption about how executable packages are handled.

Alternatively, you could add some unit tests for `runpy._get_module_details` directly to `Lib/test/test_runpy.py` that ensure that this behaviour is retained (that approach would also be a good foundation for potentially converting this to a public API in 3.8 that not only returns the current module details, but also the values to use to update the module namespace prior to execution)

Reply:
I've duplicated the test (added one for the plain modules situation).
If you find it appropriate I can add a helper function to create those "scripts" as it is quite duplicated and change the existing code to use it.

Additionally, if you find interesting the idea of making _get_module_details public I can file an issue and draft something out, as it will be useful for the other PRs on running as module